sdk/core/Azure.Core.Experimental/src/JsonPatchDocument.cs

```diff
@@ -14,35 +17,73 @@ namespace Azure.Core
     /// </summary>
     public class JsonPatchDocument
     {
-        internal Collection<JsonPatchOperation> Operations { get; }
+        private ObjectSerializer _serializer;
+        private Collection<JsonPatchOperation> Operations { get; }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="JsonPatchDocument"/> that uses <see cref="JsonObjectSerializer"/> as the default serializer.
+        /// </summary>
+        public JsonPatchDocument() : this(new JsonObjectSerializer())
+        {
+        }
 
         /// <summary>
         /// Initializes a new instance of <see cref="JsonPatchDocument"/>
         /// </summary>
-        public JsonPatchDocument()
+        /// <param name="serializer">The <see cref="ObjectSerializer"/> instance to use for value serialization.</param>
+        public JsonPatchDocument(ObjectSerializer serializer)
         {
             Operations = new Collection<JsonPatchOperation>();
+            _serializer = serializer ?? throw new ArgumentNullException(nameof(serializer));
         }
 
         /// <summary>
         /// Appends an "add" operation to this <see cref="JsonPatchDocument"/>.
         /// </summary>
         /// <param name="path">The path to apply the addition to.</param>
         /// <param name="rawJsonValue">The raw JSON value to add to the path.</param>
-        public void AppendAdd(string path, string rawJsonValue)
+        public void AppendAddRaw(string path, string rawJsonValue)
         {
             Operations.Add(new JsonPatchOperation(JsonPatchOperationKind.Add, path, null, rawJsonValue));
         }
 
+        /// <summary>
+        /// Appends an "add" operation to this <see cref="JsonPatchDocument"/>.
+        /// </summary>
+        /// <param name="path">The path to apply the addition to.</param>
+        /// <param name="value">The value to add to the path.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
+        public void AppendAdd<T>(string path, T value, CancellationToken cancellationToken = default)
+        {
+            Operations.Add(new JsonPatchOperation(JsonPatchOperationKind.Add, path, null, Serialize(value, cancellationToken)));
+        }
+
+        private string Serialize<T>(T value, CancellationToken cancellationToken)
```

Comment:
Do we care about publics above privates for classes?

Reply:
Yeah, I'll more it around